src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/AbstractParenPadCheck.java

```diff
@@ -109,13 +110,14 @@ protected void processRight(DetailAST ast) {
         final int before = ast.getColumnNo() - 1;
         if (before >= 0) {
             final String line = getLines()[ast.getLineNo() - 1];
+            final int[] codePoints = line.codePoints().toArray();
             if (option == PadOption.NOSPACE
-                && Character.isWhitespace(line.charAt(before))
+                && CommonUtil.isCodePointWhitespace(codePoints, before)
                 && !CommonUtil.hasWhitespaceBefore(before, line)) {
                 log(ast, MSG_WS_PRECEDED, CLOSE_PARENTHESIS);
             }
             else if (option == PadOption.SPACE
-                && !Character.isWhitespace(line.charAt(before))
+                && !CommonUtil.isCodePointWhitespace(codePoints, before)
```

Comment:
minor thing - `CommonUtil.isCodePointWhitespace(codePoints, before)` can be extracted, no need to invoke it 2 times

Reply:
Done

Reply:
@strkkk please resolve item